2015-Fall/nativix/viewControler.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  ViewController.swift
+//  Calculator
+//
+//  Created by MARIA NATIVIDAD LARA DIAZ on 01/09/15.
+//  Copyright (c) 2015 Appstract Development. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    @IBOutlet weak var display: UILabel!
+    
+    var userIsInTheMiddleOfTypingANumber: Bool = false
+   
+    @IBAction func appendDigit(sender: UIButton) {
+        let digit = sender.currentTitle!
+        if userIsInTheMiddleOfTypingANumber {
+            display.text = display.text! + digit
+        } else {
+            display.text = digit
+            userIsInTheMiddleOfTypingANumber = true
+        }
+    }
+    
+    var numberStack = Array<Double>()
+    var symbolStack = Array<String>()
+    
+    @IBAction func operate(sender: UIButton) {
+        userIsInTheMiddleOfTypingANumber = false
+        numberStack.append((display.text! as NSString).doubleValue)
+        println("numberStack = \(numberStack)")
+        if symbolStack.count >= 1 && numberStack.count >= 1 {
+            let operation = symbolStack.removeLast()
+            switch operation {
+            case "+" : performOperation{ $0 + $1}
```

Comment:
As shown in the lecture, some of these operations are already declared. I.e. the addition is already defined and you can simply pass in `+` to `performOperation`